### PR TITLE
[WIP] New emergency shuttle: Sanctus

### PR DESCRIPTION
**Pull request by coiax**
- New emergency shuttle, will be pricey. Called Sanctus.
- Gonna be designed like one of those pure temples, with a fountain/pool in the middle, everything all marble and stuff.
- Thanks to magical statues, the shuttle will be immune from explosions, and everyone on board will be unable to be harmed

Currently need to work out what method of no harm to implement, using the GODMODE flag also prevents healing. Could use GODMODE and constant full_heal() procs, I'm not sure.

Hopefully I'll be able to get some proper spriting and design work for this shuttle, I think it'll be very good if I can get it all artistic and shit. But the rough design will be fun.

### CODE REVIEW

File: code/game/machinery/dampener.dm [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	for(var/d in dampener_list)
+		var/obj/machinery/dampener/D = d
+		already_protected |= D.protected_turfs
+
+	protected_turfs.Cut()
+
+	var/turf/current = get_turf(src)
+
+	for(var/t in RANGE_TURFS(radius, current))
+		if(t in already_protected)
```
Comment (RemieRichards) [inline code/game/machinery/dampener.dm]: dont use ```in```, use an assoc list of ```t = TRUE```, and then just ```if(already_protected[t])```

**Comment by ShadowDimentio** [comment]
Make it so if you smash all the statues everyone gets gibbed.

**Comment by coiax** [comment]
Statues will not be destructible. That would be silly.

File: code/game/machinery/dampener.dm [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+/obj/machinery/dampener/proc/turn_off()
+	active = FALSE
+	unprotect_turfs()
+	protected_turfs.Cut()
+	update_icon()
+
+/obj/machinery/dampener/fountain
+	name = "fountain of shielding"
+	desc = "He drove out the man, and at the east of the garden of \
```
Comment (ChangelingRain) [inline code/game/machinery/dampener.dm]: hello i'm here to yell about line breaks

**Comment by Okand37** [comment]
It sounds like a fun idea, but not being able to harm anyone would render hijack nearly impossible short of teleporting or slipping and shoving them out. Perhaps you could have it gradually heal people with chemicals?
Personally I'd like to at the least have some way you could turn off the healing/godmode ability as it does bring about issues in terms of objectives, which while not a big deal for me personally, some people might get rather upset about it!
Overall, sounds interesting and I can't wait to see it!

**Comment by XDTM** [comment]
Definitely needs to be very pricey. Invest in cargo to be safe in retirement!

**Comment by MrStonedOne** [comment]
Feature.

**Comment by CPTANT** [comment]
Uh, no godmode on the shuttle is a terrible idea.

It being bombproof is ok though.

**Comment by dannno** [comment]
godmode is just not a good idea

**Comment by XDTM** [comment]
Why not tho, i like the idea of seeing wannabe griefers being forced to not hurt/bomb/flashbang people.

**Comment by PKPenguin321** [comment]
The hugbox shuttle

**Comment by coiax** [comment]
I was going to call it Hugbox, but I figured it would be too on the nose.

**Comment by KorPhaeron** [comment]
Sounds comfy

**Comment by xxalpha** [comment]
It sounds comfy until you realize people will be beating each other up the entire ride because no harm.

**Comment by coiax** [comment]
I have some ideas about how to stop that. Secret ideas. ;)

**Comment by calzilla1** [comment]
Emagging the shuttle makes it turn evil and starts hurting everyone

**Comment by improvedname** [comment]
Aka deny haijckers

**Comment by optimumtact** [comment]
This is obviously a feature, please stop making PR's we can't merge till next month

**Comment by Armhulen** [comment]
make the fountain into healing water from half life in the center of the shuttle, and the shuttle being bomb proof

invincibility is too much

**Comment by coiax** [comment]
@optimumtact I wasn't expecting it to get merged during the freeze. But I gotta work on something that isn't my dreaded cloning rework.

**Comment by PKPenguin321** [comment]
can the thing you work on be not this

**Comment by optimumtact** [comment]
If you're gonna make this a shuttle, add some interesting preconditions for it being unlocked. Like having to travel to a space ruin with the fountain of youth and bring back the cup of youth and analyze it to unlock the power of life.

the whole points thing is bogus and played out.

**Comment by KorPhaeron** [comment]
Frozen

**Comment by PKPenguin321** [comment]
Kor we have a feature frozen tag use it

**Comment by optimumtact** [comment]
closing them is the right thing to do

**Comment by optimumtact** [comment]
feature frozen tag is just a warning yoru pr is going to be closed soon

**Comment by PKPenguin321** [comment]
I thought we used it so we could sort by that tag when the freeze ends to reopen frozen PRs

**Comment by optimumtact** [comment]
Well that too

**Comment by KorPhaeron** [comment]
Lock this behind something other than credits, there is a proc for it

**Comment by coiax** [comment]
I'm just going to make it adminonly for now.

**Comment by KorPhaeron** [comment]
@coiax 

Are you gonna finish this?

**Comment by coiax** [comment]
I am hitting a design wall when it comes to the shuttle. I will probably at least make a PR for the explosion proofing mechanic, and build from there.

**Comment by KorPhaeron** [comment]
Can't you just make ex_act return nothing on your objects?

**Comment by coiax** [comment]
So, I COULD make a special turf that doesn't do anything when ex_act'd, but it would limit the design potential of the shuttle itself. Making it a flag is very simple and quite easy.

**Comment by KorPhaeron** [comment]
Just use an indestructible turf and then set the icon state in the map maker

**Comment by PKPenguin321** [comment]
i still think a shuttle that just makes you not able to die is lame

**Comment by KorPhaeron** [comment]
Well let me know when it's ready, I was looking forward to this

**Pull request closed without merging**